modin/engines/base/frame/data.py

```diff
@@ -854,20 +854,28 @@ def _fold(self, axis, func):
             self._column_widths,
         )
 
-    def filter_full_axis(self, axis, func):
+    def filter_full_axis(self, axis, func, build_map_reduce=None):
```

Comment:
Why is the `build_map_reduce` needed? Is it because of the need to wrap the result in `DataFrame`?

Reply:
Yes, it is need to wrap result in `DataFrame` because later `self._frame_mgr_cls.get_indices(0, new_partitions, lambda df: df.index)` is called but `df` won't have `index` attribute if it is `numpy.ndarray`.

Reply:
I think we can avoid this, see suggested change.